core/platform/search/gae_search_services_test.py

```diff
@@ -461,6 +466,11 @@ def test_search_all_documents(self):
             'id': 'doc3', 'k': 'abc jkl ghi', 'rank': 3, 'language_code': 'en'
         }, result)
 
+    def test_search_when_query_string_is_invalid(self):
+        result = gae_search_services.search('NOT:abc', 'my_index')
+        self.assertEqual(result, ([], None))
```

Comment:
How do you know that this is due to an invalid query string rather than "no results"?

Reply:
done

Reply:
How is this done? Can you explain this please?

Reply:
The search result would be ([], None) if query strings contain "NOT" or a string with backslashes. See [here](https://github.com/oppia/oppia/blob/develop/core/platform/search/gae_search_services.py#L298)

Reply:
But wouldn't query for invalid strings also result in an empty output?

Reply:
Yes the strings with backslashes and "NOT" are considered invalid that's why it returns an empty output. I have added another test that contains query string with a backslash

Reply:
I still don't understand what is going on here. I think you are basing this on a comment in the original file, but I would do some tests to ensure that that comment is accurate. Can you double-check that searching (on the local dev server) with those constructs actually triggers the code in the exception clause?

What you should also be testing in this test, in order to ensure that the issue is not just that there are no docs that fulfill the search query, is that logging.exception actually gets triggered as well. You can do a self.swap on logging.exception for this -- the patterns for that already exist in some of the search tests.

Reply:
I did a self.swap on logging.exception and the patterns match!